volatility3/framework/plugins/windows/unloadedmodules.py

```diff
@@ -116,53 +112,86 @@ def list_unloadedmodules(
             object_type=unloaded_count_type, offset=last_unloadedmodule_offset
         )
 
+        # Bring down to default when smear present. Some samples had this completely broken
+        if unloaded_count > 1024:
+            vollog.warning(
+                f"Smeared array count found {unloaded_count}. Defaulting to 1024 elements."
+            )
+            unloaded_count = 1024
+
         unloadedmodules_array = context.object(
             object_type=unloadedmodule_table_name
             + constants.BANG
             + "_UNLOADED_DRIVERS",
-            layer_name=layer_name,
+            layer_name=ntkrnlmp.layer_name,
             offset=unloadedmodules,
         )
         unloadedmodules_array.UnloadedDrivers.count = unloaded_count
 
+        kernel_space_start = modules.Modules.get_kernel_space_start(
+            context, kernel_module_name
+        )
+
+        address_mask = context.layers[ntkrnlmp.layer_name].address_mask
+
         for driver in unloadedmodules_array.UnloadedDrivers:
             # Mass testing led to dozens of samples backtracing on this plugin when
             # accessing members of modules coming out this list
             # Given how often temporary drivers load and unload on Win10+, I
             # assume the chance for smear is very high
             try:
-                driver.StartAddress
-                driver.EndAddress
-                driver.CurrentTime
-                yield driver
+                start_address = driver.StartAddress & address_mask
+                end_address = driver.EndAddress & address_mask
+                current_time = driver.CurrentTime
+                driver_name = driver.Name.String
             except exceptions.InvalidAddressException:
                 continue
 
+            if (
```

Comment:
How often do we fail these sanity tests?  Should we include a debug log so that we can tell when something unusual happens or are these too common to warrant that?

Reply:
No, this array is awful with smear because of how often drivers load and unload. Also, its not a full driver object, so if any of the name/start/end are busted then its just broken and there isn't anything else useful to get - probably got overwritten during acquistion.